pkg/personname/info.go

```diff
@@ -0,0 +1,150 @@
+/*
+The personName package provides methods and data types for inspecting Person Name (PN)
+Value Representations, as defined here:
+
+http://dicom.nema.org/dicom/2013/output/chtml/part05/sect_6.2.html
+*/
+package personname
+
+import (
+	"strings"
+)
+
+const groupSep = "="
+
+// Info holds information from an element with a "PN" VR. See the "PN"
+// entry at: http://dicom.nema.org/dicom/2013/output/chtml/part05/sect_6.2.html
+//
+// PN values are split into three groups which represent three different ways to
+// represent a name:
+//
+// - Alphabetic: How a name is formally spelled using a Phonetic alphabet.
+// - Ideographic: How a name is represented using ideograms / ideographs.
+// - Phonetic: How a name is pronounced.
+//
+// Each of these groups can be inspected to access their individual segments (family
+// name, Given name, etc.)
+type Info struct {
+	// Alphabetic group information about the Alphabetic group.
+	Alphabetic GroupInfo
+	// Ideographic group information about the Ideographic group.
+	Ideographic GroupInfo
+	// Phonetic group information about the Phonetic group.
+	Phonetic GroupInfo
+
+	// NoNullSeparators will remove repeated separators around null groups when
+	// calling String() if set to true.
+	NoNullSeparators bool
+}
+
+// WithNullSeparators returns a new Info object that will keep trailing separators
+// that surround both null groups AND group segments: (ex: 'Potter^Harry^^^==').
+func (info Info) WithNullSeparators() Info {
+	info.NoNullSeparators = false
+	info.Alphabetic.NoNullSeparators = false
+	info.Ideographic.NoNullSeparators = false
+	info.Phonetic.NoNullSeparators = false
+
+	return info
+}
+
+// WithoutNullSeparators returns a new Info object that will remove trailing
+// separators that surround both null groups AND group segments:
+// (ex: 'Potter^Harry').
+func (info Info) WithoutNullSeparators() Info {
+	info.NoNullSeparators = true
+	info.Alphabetic.NoNullSeparators = true
+	info.Ideographic.NoNullSeparators = true
+	info.Phonetic.NoNullSeparators = true
+
+	return info
+}
+
+// String returns the original Raw representation of the PN value, in
```

Comment:
```suggestion
// String returns a Raw representation of the PN value, in
```
suggestion: maybe not say the original representation, as it might be slightly modified based on NoNullSeperators.